manual/src/refman/extensions/tail_mod_cons.etex

```diff
@@ -0,0 +1,548 @@
+(Introduced in OCaml 4.14)
+
+Note: this feature is considered experimental, and its interface may
+evolve, with user feedback, in the next few releases of the language.
+
+Consider the simple way of implementing the "List.map" function:
+\begin{verbatim}
+let rec map f li =
+  match li with
+  | [] -> []
+  | x :: xs ->
+    let y = f x in
+    y :: map f xs
+\end{verbatim}
+
+A well-known limitation of this implementation is that the recursive
+call, "map f xs", is not in \emph{tail} position. The runtime needs to
+remember to continue with "y :: _" after the call returns and, as
+a result, this function consumes some amount of call-stack space on
+each recursive call. The stack usage of "map f li" is proportional to
+the length of "li". This is a correctness issue for large lists on
+operating systems with limited stack space -- the dreaded
+"Stack_overflow" exception.
+
+\begin{verbatim}
+# List.length (map Fun.id (List.init 1_000_000 (fun i -> i)));;
+Stack overflow during evaluation (looping recursion?).
+\end{verbatim}
+
+In this implementation of "map", the recursive call happens in
+a position that is not a \emph{tail} position in the program, but
+within a datatype constructor application that is itself in
+\emph{tail} position. We say that those positions, that are composed
+of tail positions and constructor applications, are \emph{tail modulo
+constructor} positions.
```

Comment:
it is a good place to introduce `TMC` acronym:

```tex
... of tail positions and constructor applications, are \emph{tail modulo
 constructor} (TMC) positions.
 ```

Reply:
Good point, done.